### ISSUE TITLE
r/azurerm_application_insights - provider feature disable_generated_rule isn't working

### ISSUE THREAD

**Issue by t3mi** [hashicorp/terraform-provider-azurerm#16124]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version

Terraform v1.0.11
on windows_amd64
+ provider registry.terraform.io/hashicorp/azurerm v3.0.2
### Affected Resource(s)

* `azurerm_application_insights`

### Terraform Configuration Files

```hcl
provider "azurerm" {
  features {
    application_insights {
      disable_generated_rule = true
    }
  }
}

resource "azurerm_resource_group" "example" {
  name     = "tf-test"
  location = "West Europe"
}

resource "azurerm_application_insights" "example" {
  name                = "tf-test-appinsights"
  location            = azurerm_resource_group.example.location
  resource_group_name = azurerm_resource_group.example.name
  application_type    = "web"
}

```

### Expected Behaviour

No errors, application insights resource provisioned without default rules.
### Actual Behaviour

```hcl
azurerm_application_insights.example: Still creating... [58m24s elapsed]
azurerm_application_insights.example: Still creating... [58m34s elapsed]
azurerm_application_insights.example: Still creating... [58m44s elapsed]
azurerm_application_insights.example: Still creating... [58m54s elapsed]
azurerm_application_insights.example: Still creating... [59m4s elapsed]
azurerm_application_insights.example: Still creating... [59m14s elapsed]
azurerm_application_insights.example: Still creating... [59m24s elapsed]
azurerm_application_insights.example: Still creating... [59m34s elapsed]
azurerm_application_insights.example: Still creating... [59m44s elapsed]
azurerm_application_insights.example: Still creating... [59m54s elapsed]
azurerm_application_insights.example: Still creating... [1h0m4s elapsed]
azurerm_application_insights.example: Still creating... [1h0m14s elapsed]
╷
│ Error: making Read request for Action Group: (Name "Application Insights Smart Detection" / Resource Group "tf-test"): insights.ActionGroupsClient#Get: Failure sending request: StatusCode=0 -- Original Error: context deadline exceeded
│
│   with azurerm_application_insights.example,
│   on main.tf line 14, in resource "azurerm_application_insights" "example":
│   14: resource "azurerm_application_insights" "example" {
│
```
### Steps to Reproduce

1. `terraform apply`
2. Wait default timeout (1h) to see the errors.

### References

* #10563

**Comment by katbyte** [comment]
(introduced by pr #15892)

**Comment by sebader** [comment]
Ran into the same. It actually looks like the AppInsights resource gets properly created but TF doesn't realize that

**Comment by mbfrahry** [comment]
Hey @t3mi and @sebader, the change implemented in #15892 tries to disable both the Rule and Action Group that are created by Azure when spinning up an Application Insight resource (See #10563 for more context). Sometimes, that Action Group takes forever to spin up so we end up hitting our own create timeout just waiting to disable that Action Group. 

With that said, I believe just disabling the Rule should be enough as the Action Group is just acting on that Azure created rule and nothing else. Is that true? And if so, would it make sense to just disable the Rule and leave the Action Group alone?

**Comment by sebader** [comment]
@mbfrahry I just tested this in 3.1: So the deployment works now and indeed, the rules are not created anymore, the action group gets created, still. This still doesn't solve the actual problem with the new 3.x version: the default behavior of `prevent_deletion_if_contains_resources = true`. Resource group deletion will fail since that action group is automatically created and not deleted when deleting appinsights
